sway/commands/input/xkb_switch_layout.c

```diff
@@ -99,6 +99,11 @@ struct cmd_results *input_cmd_xkb_switch_layout(int argc, char **argv) {
 			&actions[actions_len++];
 
 		action->keyboard = wlr_keyboard_from_input_device(dev->wlr_device);
+		if (action->keyboard->keymap == NULL && dev->is_virtual) {
```

Comment:
Can we maybe leave it the `dev->is_virtual` bit, and add a comment saying that no keymap might happen with virtual devices?